gobblin-yarn/src/main/java/gobblin/yarn/YarnHelixUtils.java

```diff
@@ -199,4 +206,48 @@ public static void addFileAsLocalResource(FileSystem fs, Path destFilePath, Loca
 
     return environmentVariableMap;
   }
-}
+
+  public static Optional<Integer> takePort(int desiredPort) {
+    if (desiredPort > 0) {
+      for (int port = desiredPort; port < 65535; port++) {
+        try (ServerSocket ignored = new ServerSocket(port)) {
+          return Optional.of(port);
+        } catch (IOException ignored) {
+        }
+      }
+    }
+    return Optional.absent();
+  }
+
+  public static String getJvmInputArguments() {
```

Comment:
Can you add javadoc? Especially to clarify what `getJvmInputArguments` does vs `getJvmArguments`